### ISSUE TITLE
Indexing never finishes when using `atom -a`

### ISSUE THREAD

**Issue by rahatarmanahmed** [atom/atom#12000]
### Prerequisites

* [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
* [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
* [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

The only other mention of this bug I found was here https://github.com/atom/atom/issues/10286#issuecomment-216063073, but it seems that user didn't open an issue for it.

### Description

When adding a folder to an existing atom window with `atom -a [directory]`, Atom gets stuck indexing. That is, trying to open a file with `Cmd+P` will just show this:

![users rahat downloads temp 2016-06-17 19-55-45](https://cloud.githubusercontent.com/assets/3174006/16168161/874d70d8-34c5-11e6-98ac-f5bc84d1c14f.png)

### Steps to Reproduce

1. Setup minimal test case

```sh
mkdir dir1 dir2
touch dir1/file1.txt dir2/file2.txt
```

2. Open first safe atom window (make sure atom isn't running at this point)

```sh
atom --safe dir1
```

At this point, indexing works fine and you can use `Cmd+P` as expected.

3. Add a project to the window

```sh
atom -a dir2
```

**Expected behavior:** For the indexing to finish and files of both projects to be searchable.

**Actual behavior:** The popup says `Indexing project...` forever, like in the above screenshot

### Versions

```sh
atom --version
Atom    : 1.8.0
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```

**Comment by rahatarmanahmed** [comment]
Oh, this is the wrong repo. Sorry, i'm sorry, trying to remove it.

**Comment by simurai** [comment]
@rahatarmanahmed at least you won the "who will create the 12'000th issue". :smile:

**Comment by rahatarmanahmed** [comment]
Heh. For future reference, this really *was* the right repo, and this issue is fixed in #12006
